src/vm/builtins.rs

```diff
@@ -225,19 +225,19 @@ impl<T: 'static> Forth<T> {
     }
 
     pub fn list_dict(&mut self) -> Result<(), Error> {
-        let Self {
-            run_dict_tail,
-            output,
-            ..
-        } = self;
+        let Self { output, dict, .. } = self;
         output.write_str("dictionary: ")?;
-        let mut cur = *run_dict_tail;
-
-        while let Some(item) = cur.take() {
-            let item = unsafe { item.as_ref() };
-            output.write_str(item.hdr.name.as_str())?;
+        for item in dict.entries() {
+            output.write_str(unsafe { item.entry().as_ref() }.hdr.name.as_str())?;
+            if let DictLocation::Parent(_) = item {
+                // indicate that this binding is inherited from a parent.
+                // XXX(eliza): i was initially gonna add "(inherited)" but that
+                // makes some of the tests overflow their output buffer, lol.
+                // idk if the output of the `dict` word is standardized, is this
+                // okay to do?
+                output.write_str("*")?;
```

Comment:
@jamesmunns do you have an opinion on this?

Reply:
Nope! This is just a purely invented-by-me debugging util. No format stability.